enterprise/course_catalog_api.py

```diff
@@ -64,6 +64,30 @@ def get_all_catalogs(self):
         """
         return self._load_data('catalogs', default=[])
 
+    def get_single_catalog_courses(self, catalog_id):
+        """
+        Return the courses included in a single course catalog by ID.
+
+        Args:
+            catalog_id (int): The catalog ID we want to retrieve.
+
+        Returns:
+            dict: Courses of the catalog in question
+        """
+        return self._load_data('catalogs/{catalog_id}/courses'.format(catalog_id=catalog_id))
+
+    def get_single_course_details(self, course_key):
+        """
+        Return the details of a single course by key - not a course run key.
+
+        Args:
+            course_key (str): The unique key for the course in question.
+
+        Returns:
+            dict: Details of the course in question.
+        """
+        return self._load_data('courses', resource_id=course_key, many=False)
```

Comment:
@mattdrayer: No. As above, the edX API utils will give us an empty result if the request fails.

@pomegranited: https://github.com/edx/edx-platform/blob/master/openedx/core/lib/edx_api_utils.py#L17-L18